### PR TITLE
Hashed data - samba_du, scim, sympa*

### PR DESCRIPTION
**Pull request by Johaney-s**
* these services now use hashedData

### CODE REVIEW

File: gen/sympa [hunk 1]
```diff
@@ -57,7 +51,7 @@ for my $listName (keys %$mailinglistStruc) {
     open FILE,">$fileName" or die "Cannot open $fileName: $! \n";
 
     for my $mail (keys %{$mailinglistStruc->{$listName}}) {
```
Comment (metodej) [inline gen/sympa:53]: I think we could add `sort` here to have unified output just like in the `symba_cesnet`.

File: gen/scim [hunk 2]
```diff
@@ -70,16 +70,16 @@ foreach my $uid (sort keys %$userStruc) {
...
 
 # PREPARE GROUPSDATA TO JSON
 my @groups;
 foreach my $gid (sort keys %$groupStruc) {
         my $group = {};
         $group->{"id"} = $gid;
-        $group->{"name"} = $groupStruc->{$gid}->{$A_GROUP_NAME};
-        $group->{"parentGroupId"} = $groupStruc->{$gid}->{$A_GROUP_PAR_ID};
+        $group->{"name"} = $data->getGroupAttributeValue(group => $gid, attrName => $A_GROUP_NAME);
+        $group->{"parentGroupId"} = $data->getGroupAttributeValue(group => $gid, attrName => $A_GROUP_PAR_ID);
```
Comment (metodej) [inline gen/scim:82]: I think you can get the attributes from the `groupStruc` just like it used to do or you don't have save it in there during `processGroups` method.

**Review by metodej — CHANGES_REQUESTED** [review]
The scripts look great, I have just some comments.

File: gen/scim [hunk 3]
```diff
@@ -140,22 +139,20 @@ sub processUsers {
...
-
-        if ($groupAttributes{$A_GROUP_NAME}) {
-                my $groupName = $groupAttributes{$A_GROUP_NAME};
-                my $gid = $groupAttributes{$A_GROUP_ID};
-                my $groupParId = $groupAttributes{$A_GROUP_PAR_ID};
-
-                unless(exists $groupStruc->{$gid}) {
-                        $groupStruc->{$gid}->{$A_GROUP_NAME} = $groupName;
-                        $groupStruc->{$gid}->{$A_GROUP_PAR_ID} = $groupParId;
+        my $groupId = shift;
```
Comment (balcirakpeter) [inline gen/scim:142]: bad indentation

**Review by balcirakpeter — APPROVED** [review]
The code seems to be ok. Just a small thing. It would be better to be consistent in parameter ordering while calling `getAttributeValues` subs. It will be more readable.

File: gen/scim (hunk 3 above)
Comment (Johaney-s) [reply gen/scim]: fixed

File: gen/scim (hunk 2 above)
Comment (Johaney-s) [reply gen/scim]: You're right, kept the old approach.

File: gen/sympa (hunk 1 above)
Comment (Johaney-s) [reply gen/sympa]: added

**Review by metodej — APPROVED** [review]

File: gen/sympa [hunk 4]
```diff
@@ -25,29 +25,23 @@ our $A_USER_STATUS;                      *A_USER_STATUS =                     \'
...
-
-    my @membersData = $rData->getChildElements;
+foreach my $resourceId ($data->getResourceIds()) {
+	my $listName = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_NAME);
 
-    my $listName = "$resourceAttributes{$A_RESOURCE_MAILING_LIST_NAME}";
-
-    foreach my $memberAttributes (dataToAttributesHashes @membersData) {
+	for my $memberId ($data->getMemberIdsForResource(resource => $resourceId)) {
         #list only VALID members except if allowNonvalidUsers is true
```
Comment (zlamalp) [inline gen/sympa:38]: I'ts not your change, but this comment doesn't make sense and its probably copy-paste from original mailman script. We do not check any resource attribute to allow non-valid users, script forces usage of only valid users by the next command. Please remove the comment. Same goes for `sympa_cesnet` version of the service.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: gen/sympa (hunk 4 above)
Comment (Johaney-s) [reply gen/sympa]: Removed.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**